### DISCUSSION TITLE
Support Object Storage *prefix* in format command so that we can restrict JuiceFs under a specific prefix

### DISCUSSION THREAD

**Discussion by ericsun2** [juicedata/juicefs#4904] [General]
**What would you like to be added**:

```
juicefs format --storage s3 \
    --bucket https://myjuicefs.s3.us-east-2.amazonaws.com/prefix_to_juicefs/project_01 \
    redis://192.168.1.6/1 \
    myjfs
```
or
```
juicefs format --storage s3 \
    --bucket https://myjuicefs.s3.us-east-2.amazonaws.com \
    --prefix prefix_to_juicefs/project_01 \
    redis://192.168.1.6/1 \
    myjfs
```

**Why is this needed**:
In some case, we can restrict JuiceFs to operate within a given `prefix` in the Object Storage. The above S3 example should be applicable to GCS, Azure, ... as well. `s3://myjuicefs/prefix_to_juicefs/project_01` will be the root location for the `format` command to create a volume.

Object Storage ACL can also be applied to the `prefix`.

**Comment by davies** [comment]
right now, we use volume name as the prefix, why that is not enough?

**Comment by is** [comment]
There is indeed a need for this

**Comment by 1247862674** [comment]
I now want to format the S3 prefix (/project/project1/userdata). What do I need to do?
